lib/src/working_copy.rs

```diff
@@ -1228,6 +1232,9 @@ impl TreeState {
          -> Result<(), CheckoutError> {
             let disk_path = path.to_fs_path(&self.working_copy_path);
 
+            if before.is_present() {
+                fs::remove_file(&disk_path).ok();
+            }
```

Comment:
redundant `fs::remove_file()` in `Ok(None)` case

Reply:
Fixed, thanks.